### ISSUE TITLE
TencentAILab/tensorflow not available

### ISSUE THREAD

**Issue by xueyouluo** [TencentAILab/tf-adaptive-softmax-lstm-lm#4]
so TencentAILab/tensorflow is not open-sourced?  where can i find tf.contrib.layers.adaptive_softmax_loss?

**Comment by Froskekongen** [comment]
Any news on this?

**Comment by yangsaiyong** [comment]
The adaptive softmax impentation has been extracted from tf.contrib.layers.adaptive_softmax_loss, and was wrapped in `softmax.py`. (No other dependencies required.)

**Comment by yangsaiyong** [comment]
The adaptive softmax implementation has been extracted from tf.contrib.layers.adaptive_softmax_loss, and was wrapped in softmax.py. (No other dependencies required.)
